### ISSUE TITLE
Naming conflict when replacing default relations

### ISSUE THREAD

**Issue by alaister** [graphile-contrib/pg-many-to-many#38]
Given the following schema:
```sql
CREATE TABLE users (
    id uuid DEFAULT gen_random_uuid() PRIMARY KEY
);

CREATE TABLE posts (
    id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
    title text NOT NULL,
    content text,
    created_at timestamp with time zone DEFAULT now() NOT NULL
);

CREATE TABLE post_authors (
    id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
    post_id uuid NOT NULL REFERENCES posts(id),
    user_id uuid NOT NULL REFERENCES users(id),
    created_at timestamp with time zone DEFAULT now() NOT NULL
);

CREATE INDEX ON post_authors (post_id);
CREATE INDEX ON post_authors (user_id);

COMMENT ON TABLE post_authors IS E'@omit many';

COMMENT ON CONSTRAINT post_authors_user_id_fkey ON post_authors IS E'@manyToManyFieldName authors';
```
Results in the error:
```
Error: A type naming conflict has occurred - two entities have tried to define the same type 'PostAuthorsEdge'.

  The first entity was:

    Adding table edge type for table "public"."post_authors". You can rename the table's GraphQL type via:
    
      COMMENT ON TABLE "public"."post_authors" IS E'@name newNameHere\n@omit many';

  The second entity was:

    'newWithHooks' call during hook 'PgManyToManyPlugin/GraphQLObjectType:fields/unnamed'
    ...
```
I believe this is because Postgraphile still generates it's own `PostAuthorsEdge` even though I have disabled it with `@omit many`. Then renaming my many to many field name to be authors causes pg-many-to-many to generate a type of the same name.

Currently the only solution I have found is to rename the table to something different as Postgraphile suggests, but that defeats the point of renaming the many to many field in the first place!

**Comment by mattbretl** [comment]
> I believe this is because Postgraphile still generates it's own PostAuthorsEdge even though I have disabled it with @omit many. Then renaming my many to many field name to be authors causes pg-many-to-many to generate a type of the same name.

Correct. The default Edge and Connection types are created during an "init" hook, which means they'll exist even if they aren't used in the final schema, and that behavior can't be changed in core until v5. So we need to work around that.

I think I'll just add "ManyToMany" to the Edge/Connection type names generated by this plugin to ensure uniqueness. If anyone wants to shorten the type names for their specific use case, they can override the inflector, but avoiding field name conflicts on startup is a top priority.

**Comment by alaister** [comment]
That sounds like a good compromise until V5 comes around. I'm more concerned with nice field names than type names.
Thanks!

**Comment by mattbretl** [comment]
Fixed in 1.0.0-beta.12
